### PR TITLE
Document the walk_type argument default in directed_laplacian and similar functions.

### LINKED ISSUE #7147: Document walk_type in directed_laplacian and friends
The current documentation of the "walk_type" argument to the many directed laplacian matrix approaches is opaque. And the logic is hidden within the actual code because it occurs in a helper function presumably so the logic is the same for all these functions. The logic should be fairly straightforward to explain. For the directed graph case it is given in `linalg.laplacianmatrix._transition_matrix` as
```python
    if walk_type is None:
        if nx.is_strongly_connected(G):
            if nx.is_aperiodic(G):
                walk_type = "random"
            else:
                walk_type = "lazy"
        else:
            walk_type = "pagerank"
```
We should put a word description of this into the `walk_type` parameter description..

### PR DESCRIPTION
**Pull request by Ay-slim**
Closes #7147

### CODE REVIEW

File: networkx/linalg/laplacianmatrix.py [hunk 1]
```diff
@@ -219,8 +219,13 @@ def directed_laplacian_matrix(
...
     walk_type : string or None, optional (default=None)
-       If None, `P` is selected depending on the properties of the
-       graph. Otherwise is one of 'random', 'lazy', or 'pagerank'
+       If None, the value of walk_type is set based on the following conditions:
+        If nx.is_strongly_connected(G) and nx.is_aperiodic(G) are truthy:
+            walk_type is set to the string 'random'
+        If nx.is_strongly_connected(G) evaluates is truthy but nx.is_aperiodic(G) is falsy:
+            walk_type is set to the string 'lazy'
+        Otherwise (nx.is_strongly_connected(g) is falsy):
+            walk_type is set to the string 'pagerank'
```
Comment (rossbar) [inline networkx/linalg/laplacianmatrix.py:228]: The selection logic is correct and documenting it is a nice improvement. I'd vote to present the options first, then more concisely summarize the logic for the default case. Something like:

```suggestion
       One of ``"random"``, ``"lazy"``, or ``"pagerank"``. If ``walk_type=None``
       (the default), then a value is selected according to the properties of `G`:
       - ``walk_type="random"`` if `G` is strongly connected and aperiodic
       - ``walk_type="lazy"`` if `G` is strongly connected but not aperiodic
       - ``walk_type="pagerank"`` for all other cases.
```

Comment (Ay-slim) [reply networkx/linalg/laplacianmatrix.py]: Thanks @rossbar

**Review by dschult — COMMENTED** [review]
This looks good — thanks!
Can we also put similar language in the other functions that use `walk_type`.  Those include `directed_combinatorial_laplacian matrix` and ` _transiton_matrix` and might include others. Maybe search for walk_type.

**Comment by Ay-slim** [comment]
All done @dschult, I found walk_type as an argument to the following functions: networkx/networkx/linalg/laplacianmatrix.py/directed_laplacian_matrix
networkx/networkx/algorithms/centrality/laplacian.py/laplacian_centrality
networkx/networkx/linalg/laplacianmatrix.py/directed_combinatorial_laplacian_matrix
networkx/networkx/linalg/laplacianmatrix.py/_transition_matrix

They have been updated.

**Review by rossbar — APPROVED** [review]
That looks like all the instances to me, thanks @Ay-slim and @dschult !

**Pull request merged**